docs/src/util/Algolia.js

```diff
@@ -0,0 +1,85 @@
+const pageQuery = `{
+    allMdx {
+      edges {
+        node {
+          id
+          slug
+          frontmatter {
+            title
+            description
+            keywords
+            parentTab
+          }
+          headings {
+            value
+            depth
+          }
+        }
+      }
+    }
+  }`;
+
+const transformAlgoliaData = (pageQuery) => {
+  const result = pageQuery.allMdx.edges.map(({ node: { id, headings, frontmatter, slug, ...rest } }) => {
+    let customTitle = '';
+    let customDescription = '';
+    let customKeywords = '';
+    let customParentTab = '';
+    const customHeadings = [];
+
+    const originalTitle = frontmatter.title;
+    const originalDescription = frontmatter.description;
+    const originalKeywords = frontmatter.keywords;
+    const originalParentTab = frontmatter.parentTab;
+
+    if (originalTitle !== originalDescription) {
+      customDescription = originalDescription;
+      customTitle = originalTitle;
+      customKeywords = originalKeywords;
+      customParentTab = originalParentTab;
+
+      const headingMapping = headings.map((item) => {
+        if (item.depth === 4) {
+          customHeadings.push(item.value);
+        }
+      });
+
+      const subHeadingMapping = headings.map((item) => {
+        if (item.depth === 5) {
+          customHeadings.push(item.value);
+        }
+      });
```

Comment:
Please use `filter` method instead of `map`. Also "headingMapping" & "subHeadingMapping" array are not used anywhere. Please check